File: TycoonExersizes3/Route.fs
module TycoonExersizes3.Route
open TycoonExersizes3.Types
    
let nextSegment startPoint route =
    route.segments
    |> List.find (fun segment -> segment.startPoint = startPoint)
    
let targetPoint route =
    route.segments
    |> List.last
    |> fun s -> Sink s.endPoint

File: TycoonExersizes/Domain/Route.cs
using System.Linq;

namespace TycoonExersizes.Domain
{
    public class Route
    {
        public Route(RouteSegment[] segments)
        {
            Segments = segments;
        }

        public RouteSegment[] Segments { get; }
        public Point TargetPoint => Segments.Last().TargetPoint;
    }
}